### DISCUSSION TITLE
Performance on large datasets

### DISCUSSION THREAD

**Discussion by MrSquaare** [wuba/react-native-echarts#65] [General]
Hello,

`react-native-echarts` performs much better than its webview-based alternatives, but still suffers of performance issues on large datasets.

Are there any plans to improve this? Do you know whether it's the svg string conversion or the rendering that's causing the performance issues?

**Comment by zhiqingchen** [comment]
https://github.com/wuba/react-native-echarts/issues/16

Yes, this is a known issue. Using declarative ui to render charts with large amounts of data can cause large performance issues. Just like when using echarts in the web, it is also recommended to use canvas mode instead of svg mode. I haven't thought of a good solution yet, but we haven't given up on solving this problem.

One way to do this is to implement CanvasRenderingContext2D and then draw the chart by using imperative. In this case there is no need to keep a large number of nodes compared to svg. When the chart changes, it can also be optimized to redraw dirty areas instead of full rendering.

**Comment by zhiqingchen** [comment]
If you have good ideas, welcome to discuss them together
